User Interface/gui.py

```diff
@@ -0,0 +1,136 @@
+from tkinter import *
+import ipaddress
+import socket
+from tkinter import ttk
+ 
+window = Tk()
+ 
+window.title("Network & Port Scanning")
+ 
+tab_control = ttk.Notebook(window)
+ 
+port_tab = ttk.Frame(tab_control)
+ 
+net_tab = ttk.Frame(tab_control)
+ 
+tab_control.add(port_tab, text='Port Scanning')
+ 
+tab_control.add(net_tab, text='Network Scanning')
+ 
+
+#port scanning tab
+
+def scan():
+    global address
+    #print(host_address.get())
+    #print(scan_mode.get())
+    #print(host_address.get())
+    try:
+        address = ipaddress.ip_address(host_address.get())
+    
+        if(scan_mode.get() == 0):
+            print("TCP SYN Scan")
+            print("Host address = "+host_address.get())
+            print("Destination Port = "+dst_port.get())
+            print("Source Port = "+src_port.get())
+            response_text.insert(END,"TCP SYN Scan result")
+        elif(scan_mode.get() == 1):
+            print("TCP Connect Scan")
+            print("Host address = "+host_address.get())
+            print("Destination Port = "+dst_port.get())
+            print("Source Port = "+src_port.get())
+        elif(scan_mode.get() == 2):
+            print("TCP NULL Scan")
+            print("Host address = "+host_address.get())
+            print("Destination Port = "+dst_port.get())
+            print("Source Port = "+src_port.get())
+        elif(scan_mode.get() == 3):
+            print("FIN Scan")
+            print("Host address = "+host_address.get())
+            print("Destination Port = "+dst_port.get())
+            print("Source Port = "+src_port.get())
+        elif(scan_mode.get() == 4):
+            print("Xmas Scan")
+            print("Host address = "+host_address.get())
+            print("Destination Port = "+dst_port.get())
+            print("Source Port = "+src_port.get())
+        elif(scan_mode.get() == 5):
+            print("TCP ACK Scan")
+            print("Host address = "+host_address.get())
+            print("Destination Port = "+dst_port.get())
+            print("Source Port = "+src_port.get())
+        elif(scan_mode.get() == 6):
+            print("TCP Window Scan")
+            print("Host address = "+host_address.get())
+            print("Destination Port = "+dst_port.get())
+            print("Source Port = "+src_port.get())
+        elif(scan_mode.get() == 7):
+            print("UDP Scan")
+            print("Host address = "+host_address.get())
+            print("Destination Port = "+dst_port.get())
+            print("Source Port = "+src_port.get())
+    except:
+        print('invalid host address')
```

Comment:
Let's add an error message box here too so that the user can know what went wrong (when the ip is not valid)